rust/scx_utils/src/topology.rs

```diff
@@ -439,7 +442,16 @@ fn create_insert_cpu(
 
     // Physical core ID
     let top_path = cpu_path.join("topology");
-    let core_id = read_file_usize(&top_path.join("core_id"))?;
+    let core_id_raw = read_file_usize(&top_path.join("core_id"))?;
+    let cluster_id = read_file_usize(&top_path.join("cluster_id")).unwrap_or(0);
+    let die_id = read_file_usize(&top_path.join("die_id")).unwrap_or(0);
+
+    // core_id can be reused across nodes. replace core_id with a hash of topo ids
+    let mut hasher = DefaultHasher::new();
+    core_id_raw.hash(&mut hasher);
```

Comment:
Can we just hash a (core_id_raw, cluster_id, die_id) tuple instead of passing the hasher through each usize var? Other than that this change looks strictly better than what we currently have so LGTM.

Reply:
I'm not sure about hashing. For, CPU and node IDs, there are direct kernel interfaces that consume them and we probably should keep the same IDs but for LLC and core, we can just sequentially assign IDs and then add another field - e.g. hw_id - in case hardware IDs are necessary.

For node and CPU, we should provide iteration interfaces.